src/main/java/ch/njol/skript/expressions/ExprChunk.java

```diff
@@ -18,83 +18,109 @@
  */
 package ch.njol.skript.expressions;
 
+import java.util.Arrays;
+
 import ch.njol.skript.Skript;
 import ch.njol.skript.classes.Changer.ChangeMode;
 import ch.njol.skript.doc.Description;
 import ch.njol.skript.doc.Examples;
 import ch.njol.skript.doc.Name;
 import ch.njol.skript.doc.Since;
-import ch.njol.skript.expressions.base.PropertyExpression;
 import ch.njol.skript.lang.Expression;
 import ch.njol.skript.lang.ExpressionType;
 import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
 import ch.njol.skript.util.Direction;
 import ch.njol.util.Kleenean;
 import org.bukkit.Chunk;
 import org.bukkit.Location;
+import org.bukkit.World;
 import org.bukkit.event.Event;
-import org.eclipse.jdt.annotation.Nullable;
+import org.jetbrains.annotations.Nullable;
 
-/**
- * @author Peter Güttinger
- */
 @Name("Chunk")
-@Description("The <a href='./classes.html#chunk'>chunk</a> a block, location or entity is in.")
-@Examples("add the chunk at the player to {protected chunks::*}")
-@Since("2.0")
-public class ExprChunk extends PropertyExpression<Location, Chunk> {
+@Description("Returns the <a href='./classes.html#chunk'>chunk</a> of a block, location or entity is in, or a list of the loaded chunks of a world.")
+@Examples({
+	"add the chunk at the player to {protected chunks::*}",
+	"set {_chunks::*} to the loaded chunks of the player's world"
+})
+@Since("2.0, INSERT VERSION (loaded chunks)")
+public class ExprChunk extends SimpleExpression<Chunk> {
 	
 	static {
-		Skript.registerExpression(ExprChunk.class, Chunk.class, ExpressionType.PROPERTY, "[the] chunk[s] (of|%-directions%) %locations%", "%locations%'[s] chunk[s]");
+			Skript.registerExpression(ExprChunk.class, Chunk.class, ExpressionType.COMBINED,
+				"[all [of]] [the] chunk[s] (of|%-directions%) %locations%",
+				"%locations%'[s] chunk[s]",
+				"[all [of]] [the] loaded chunks (of|in) %worlds%"
```

Comment:
I don't think this pattern looks quite right.
```
all loaded chunks (ok)
all the loaded chunks (ok)
all of the loaded chunks (ok)
all of loaded chunks (???)
the loaded chunks (ok)
```
You could try something like `(all [[of] the]|the|)` instead to eliminate the bad `all of chunks` pattern.